examples/ex40.cpp

```diff
@@ -0,0 +1,350 @@
+//                                MFEM Example 40
+//
+// Compile with: make ex40
+//
+// Sample runs: ex40 -step 10 -gr 2.0
+//              ex40 -step 10 -gr 2.0 -o 3 -r 1
+//              ex40 -step 10 -gr 2.0 -r 4 -m ../data/l-shape.mesh
+//
+// Description: This example code demonstrates how to use MFEM to solve the
+//              eikonal equation,
+//
+//                      |∇u| = 1 in Ω,  u = g on ∂Ω.
+//
+//              This example constructs a fast converging sequence,
+//
+//                      uₖ → u  as k → \infty,
+//
+//              by using in Newton's method to solve the sequence of nonlinear
+//              saddle-point problems
+//
+//               Find ψₖ ∈ H(div,Ω) and uₖ ∈ L²(Ω) such that
+//               ( Zₖ(ψₖ) , τ ) + ( uₖ , ∇⋅τ ) = 0                    ∀ τ ∈ H(div,Ω)
+//               ( ∇⋅ψₖ , v )                 = ( -1 + ∇⋅ψₖ₋₁ , v )   ∀ v ∈ L²(Ω)
```

Comment:
Please explain how the BC `u=g` affects this formulation. I couldn't find anything in the code either, does the example assume `g=0`?